src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -216,6 +216,140 @@ namespace AppInstaller::CLI::Workflow
+            {
+                std::vector<std::string> preference;
+                std::vector<std::string> requirement;
+
+                // Preference will come from winget settings or Preferred Languages settings. winget settings takes precedence.
+                preference = Settings::User().Get<Settings::Setting::InstallLocalePreference>();
+                if (preference.empty())
+                {
+                    preference = Locale::GetUserPreferredLanguages();
+                }
+
+                // Requirement may come from args or settings; args overrides settings.
+                if (args.Contains(Execution::Args::Type::Locale))
+                {
+                    requirement.emplace_back(args.GetArg(Execution::Args::Type::Locale));
+                }
+                else
+                {
+                    requirement = Settings::User().Get<Settings::Setting::InstallLocaleRequirement>();
+                }
+
+                if (!preference.empty() || !requirement.empty())
+                {
+                    return std::make_unique<LocaleComparator>(preference, requirement);
+                }
+                else
+                {
+                    return {};
+                }
+            }
+
+            bool IsApplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                if (m_requirement.empty())
+                {
+                    return true;
+                }
+
+                for (auto const& requiredLocale : m_requirement)
+                {
+                    if (Locale::GetDistanceOfLanguage(requiredLocale, installer.Locale) >= Locale::MinimumDistanceScoreAsPerfectMatch)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
+            {
+                std::string result = "Installer locale does not match required locale: ";
+                result += installer.Locale;
+                return result;
+            }
+
+            bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
+            {
+                if (m_preference.empty())
+                {
+                    return false;
+                }
+
+                for (auto const& preferredLocale : m_preference)
+                {
+                    double firstScore = first.Locale.empty() ? Locale::UnknownLanguageDistanceScore : Locale::GetDistanceOfLanguage(preferredLocale, first.Locale);
+                    double secondScore = second.Locale.empty() ? Locale::UnknownLanguageDistanceScore : Locale::GetDistanceOfLanguage(preferredLocale, second.Locale);
+
+                    if (firstScore > secondScore && firstScore >= Locale::MinimumDistanceScoreAsCompatibleMatch)

```

Comment:
>if (firstScore > secondScore && firstScore >= Locale::MinimumDistanceScoreAsCompatibleMatch) [](http://example.com/codeflow?start=20&length=92)

The search through m_preference needs to stop whenever either score is compatible.  So:
```
if (firstScore >= Compat || secondScore => Compat)
  return firstScore > secondScore;
```
This is required or you could have IsFirstBetter return true for both orderings of first and second.